### ISSUE TITLE
code question in rand_cutout

### ISSUE THREAD

**Issue by foobar8675** [lucidrains/stylegan2-pytorch#185]
in this method, 
```
def rand_cutout(x, ratio=0.5):
    cutout_size = int(x.size(2) * ratio + 0.5), int(x.size(3) * ratio + 0.5)
    offset_x = torch.randint(0, x.size(2) + (1 - cutout_size[0] % 2), size=[x.size(0), 1, 1], device=x.device)
    offset_y = torch.randint(0, x.size(3) + (1 - cutout_size[1] % 2), size=[x.size(0), 1, 1], device=x.device)
    grid_batch, grid_x, grid_y = torch.meshgrid(
        torch.arange(x.size(0), dtype=torch.long, device=x.device),
        torch.arange(cutout_size[0], dtype=torch.long, device=x.device),
        torch.arange(cutout_size[1], dtype=torch.long, device=x.device),
    )
    grid_x = torch.clamp(grid_x + offset_x - cutout_size[0] // 2, min=0, max=x.size(2) - 1)
    grid_y = torch.clamp(grid_y + offset_y - cutout_size[1] // 2, min=0, max=x.size(3) - 1)
    mask = torch.ones(x.size(0), x.size(2), x.size(3), dtype=x.dtype, device=x.device)
    mask[grid_batch, grid_x, grid_y] = 0
    x = x * mask.unsqueeze(1)
    return x
```
i'm not able to follow what this line is doing. can you help or maybe point me to a tutorial?

the reason I'm confused is `mask`, `grid_batch` and `grid_x`, `grid_y` all have 3 dimensions. so how can they index into mask?

**Comment by lucidrains** [comment]
@foobar8675 Hi Matthew, I'll have to redirect you to the repository I took the code from https://github.com/mit-han-lab/data-efficient-gans

**Comment by foobar8675** [comment]
thanks @lucidrains
